src/covariances.jl

```diff
@@ -0,0 +1,87 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    design_matrix(comps)
+    
+Converts a vector of Composition objects into a Matrix of numbers.
+"""
+function design_matrix(comps)
+    matrix = reduce(hcat, components.(comps)[:,1])'
+    return matrix
+end
+
+"""
+    get_subcomposition_matrix(matrix, entries)
+
+Returns the subcomposition matrix given the `entries` (selected columns) from the `matrix`.
+"""
+function get_subcomposition_matrix(matrix, entries)
+    subcomposition_matrix = matrix[:, entries]
+    return subcomposition_matrix
+end
+
+"""
+    comp_var_array(comps)
+
+Returns the compositional variation array, from definition 4.3 of Aitchson - The Statistical Analysis of Compositional Data.
+"""
+function comp_var_array(comps)
+    dm = design_matrix(comps)
+    N, D = size(dm)
+    cva = zeros(D, D)
```

Comment:
This is not necessary and expensive in general. We need to:

```julia
for (i, c) in enumerate(comps)
  x = components(c)
  for j in i+1:D
    # do something and update tau[i,j]
  end
end
```

Reply:
Maybe it actually is fine. We can create the design matrix and access columns in the loop, otherwise the code will look not very legible.

Reply:
Got it. I'm currently working on it.